planner/core/find_best_task.go

```diff
@@ -469,6 +469,30 @@ func (ds *DataSource) findBestTask(prop *property.PhysicalProperty) (t task, err
 				p: dual,
 			}, nil
 		}
+		if !ds.isPartition && len(path.Ranges) > 0 {
+			allRangeIsPoint := true
+			for _, ran := range path.Ranges {
+				if !ran.IsPoint(ds.ctx.GetSessionVars().StmtCtx) {
+					allRangeIsPoint = false
+					break
+				}
+			}
+			if allRangeIsPoint {
+				var pointGetTask task
+				var err error
+				if len(path.Ranges) == 1 {
+					pointGetTask, err = ds.convertToPointGet(prop, candidate)
+				} else {
+					pointGetTask, err = ds.convertToBatchPointGet(prop, candidate)
+				}
+				if err != nil {
+					return nil, err
+				}
+				if pointGetTask.cost() < t.cost() {
```

Comment:
If we would always prefer PointGet, we can simply return here?

Reply:
No, we can't consider this query:

```sql
select * from t where t.a = 1 and t.b in (1, 2, 3);
```

If there is a unique index for t.a and a unique index for t.b. Then we should compare the cost for these two point-get plans.